[Gradient Compression] Update the comment on fp16_compress_hook

wayi1 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#53780 [Gradient Compression] Update the comment on fp16_compress_hook**

Update the comment, because the input data type of `fp16_compress_hook` does not have to be FP32. For example, the input dtype can also be FP64, as long as it can be casted into FP16.

Differential Revision: [D26967224](https://our.internmc.facebook.com/intern/diff/D26967224/)

Commit:
[Gradient Compression] Update the comment on fp16_compress_hook

Update the comment, because the input data type of `fp16_compress_hook` does not have to be FP32. For example, the input dtype can also be FP64, as long as it can be casted into FP16.

Differential Revision: [D26967224](https://our.internmc.facebook.com/intern/diff/D26967224/)

[ghstack-poisoned]

iseessel reviewed (approved):
Nit pick: Doesn't it take the mean before decompressing, the comment sounds otherwise?

Nit pick: "Once compressed gradient tensors are all reduced, it then calls ``decompress``, casting..."

Commit:
Update on "[Gradient Compression] Update the comment on fp16_compress_hook"


Update the comment, because the input data type of `fp16_compress_hook` does not have to be FP32. For example, the input dtype can also be FP64, as long as it can be casted into FP16.

Differential Revision: [D26967224](https://our.internmc.facebook.com/intern/diff/D26967224/)

[ghstack-poisoned]

Commit:
Update on "[Gradient Compression] Update the comment on fp16_compress_hook"


Update the comment, because the input data type of `fp16_compress_hook` does not have to be FP32. For example, the input dtype can also be FP64, as long as it can be casted into FP16.

Differential Revision: [D26967224](https://our.internmc.facebook.com/intern/diff/D26967224/)

[ghstack-poisoned]

SciPioneer commented:
> Nit pick: Doesn't it take the mean before decompressing, the comment sounds otherwise?
> 
> Nit pick: "Once compressed gradient tensors are all reduced, it then calls `decompress`, casting..."

Updated the comments.
